degica/xe_currency
# Use REST API

## Context

The gem was developed 5 years ago and it's not been updated for a long time.
In the meantime, it seems that xe.com provides the. REST API, so we should switch it.

## How to test

0. rspec pass should pass.
1. Get a credential at https://currencydata.xe.com/account/dashboard.
2. Confirm that you can get currency rates and it works with `money` gem:
    ```ruby
    require 'xe_currency'

    client = XeCurrency::Client.new(
                                    account_api_id: '<Account API ID>',
                                    account_api_key: '<Account API Key>')

    client.fetch_rate('USD', 'EUR')
    client.fetch_rates('USD', ['EUR', 'JPY'])
    ```

    ```ruby
    require 'money'
    require 'money/bank/xe_currency'

    # set default bank to instance of XeCurrency
    Money.default_bank =
      Money::Bank::XeCurrency.new(
        Money::RatesStore::Memory.new,
        account_api_id: '<Account API ID>',
        account_api_key: '<Account API Key>'
      )

    # create a new money object, and use the standard #exchange_to method
    money = Money.new(1_00, "USD") # amount is in cents
    money.exchange_to(:EUR)
    ```
